exec/Chainweb/Backfill.hs

```diff
@@ -55,19 +54,20 @@ progress count mins = do
   start <- getPOSIXTime
   forever $ do
     threadDelay 30_000_000  -- 30 seconds. TODO Make configurable?
-    total <- readIORef count
+    completed <- readIORef count
     now <- getPOSIXTime
-    let !perc = (100 * fromIntegral total / fromIntegral remaining) :: Double
-        !tilNow = (now - start) / 60
-        !tilDone0 = floor (tilNow * fromIntegral remaining / fromIntegral total) :: Int
-        (!period, !tilDone) | tilDone0 < 60 = ("minutes" :: String, tilDone0)
-                            | otherwise = ("hours", tilDone0 `div` 60)
-    printf "[INFO] Progress: %d/%d blocks (%.2f%%). ~%d %s remaining.\n"
-      total remaining perc tilDone period
+    let perc = (100 * fromIntegral completed / fromIntegral total) :: Double
+        elapsedMinutes = (now - start) / 60
+        blocksPerMinute = (fromIntegral completed / realToFrac elapsedMinutes) :: Double
+        estMinutesLeft = floor (fromIntegral (total - completed) / blocksPerMinute) :: Int
+        (timeUnits, timeLeft) | estMinutesLeft < 60 = ("minutes" :: String, estMinutesLeft)
+                              | otherwise = ("hours", estMinutesLeft `div` 60)
+    printf "[INFO] Progress: %d/%d blocks (%.2f%%), ~%d %s remaining at %.0f blocks per minute.\n"
+      completed total perc timeLeft timeUnits blocksPerMinute
```

Comment:
Could `floor` to `blocksPerMinute` here and then use `%d` in the pattern.

Reply:
I tried that.  It required an extra type annotation, so I decided this was cleaner.